src/rgw/rgw_metadata.cc

```diff
@@ -756,8 +756,19 @@ int RGWMetadataManager::put(string& metadata_key, bufferlist& bl,
   string entry;
 
   int ret = find_handler(metadata_key, &handler, entry);
-  if (ret < 0)
+  if (ret < 0) {
     return ret;
+  }
+
+  if (store->get_zone().tier_type == "archive") {
```

Comment:
This code should really only appear in the `archive` specific handlers.

Reply:
ok